enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,574 @@
+
+![Alt text](assets/community-operators.png?raw=true "community-operators")
+
+The way we describe the intent of a bundle to be on a particular channel in a package is by hardcoding that information in the [bundle annotations](https://github.com/operator-framework/operator-registry/blob/release-4.6/docs/design/operator-bundle.md#bundle-annotations) inside the bundle metadata itself. However, the channel that the bundle wants to be in is more of a package level information. Hardcoded channel information inside a bundle restricts a bundle in a particular channel once the bundle is built and loaded into an index. This makes it difficult to change channel info retroactively. When a bundle is added to an index using `opm`, the `mode` (one of `replaces`|`semver`|`semver-skippatch`) in which the bundle is added (`opm index add --bundles <list-of-bundles> --mode <replaces|semver|semver-skippatch>`) determines the upgrade graph of bundles in the channel (which bundle can be upgrade to which). This upgrade graph for bundles in a channel can also not be changed retroactively. The only way to change the channel info, the upgrade graph or any package level information currently is to rebuild the bundles with the modified package information and then republish the index. This is done by making additive changes to the index using the`opm` tool. 
+
+Furthermore, a lot of the rules about the structure of an index are implicitly documented in various sources that are scattered. A best effort is being made to keep the rules updated in the form of a library in the [api repo](https://github.com/operator-framework/api/tree/master/pkg/validation) that is imported by tools like `opm`, `operator-courier` etc that perform index validation. However, like all tools that aspire to provide data validation, these tools, or api libraries that these tools import the rules from, have to be constantly maintained to keep up with the latest rule changes, while also maintaining older rules in previous releases. This introduces a lot of overhead in ensuring that olm is easy to use by users, while also leaving windows for lapses due to mis-coordination between different tools across different versions.
+
+Since indexes are manipulated by making additive changes to it, the way an index can be reproduced today is my managing a knowledge base of the series of changes made to the index (i.e storing the `opm` commands chronologically) along with a set of intermediary artifacts (images of the index with their corresponding digest etc). This introduces a lot of overhead in maintaining indexes, and as a by product has significantly burdened the `opm` tool with feature requests to add an increasing number of sub-commands to allow index authors to manipulate their indexes in various ways. 
+
```

Comment:
so, what you are looking for here is tooling that will help you to be able to create a new bundle from a pre-existent bundle with new values for its manifests. Am I right?

So, would not easier we only update the bundle manifests (csv and/or crds) and run a command to overwrite the bundle with new name and/or version and/or channnels? E.g `operator-sdk generate bundle --input-dir=<Directory to read an existing bundle from> --overwrite=true --version=<new version>` and/or `operator-sdk generate bundle --input-dir=<Directory to read an existing bundle from> --overwrite=true --version=<new version> --package=<optional new pkg name>` --channels=<optional list of channels>` .

Reply:
@camilamacedo86 the declarative index config isn't concerned with specific operator bundles and the manifests inside them. The declarative config is for the layer that is a collection of numerous bundles, i.e the index (i.e the catalog of operators). 
What is being talked about here is the ability to create a new index(catalog of operators) from an old index(catalog of operators) without having the rebuild the old index.